modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/snapshot/daemon/SelectablePeerDiscoveryDelay.scala

```diff
@@ -0,0 +1,69 @@
+package org.tessellation.sdk.infrastructure.snapshot.daemon
+
+import cats.effect.Async
+import cats.syntax.applicative._
+import cats.syntax.applicativeError._
+import cats.syntax.eq._
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+
+import scala.concurrent.duration.FiniteDuration
+
+import org.tessellation.cli.AppEnvironment
+import org.tessellation.cli.AppEnvironment.Dev
+import org.tessellation.schema.node.NodeState.Ready
+import org.tessellation.schema.peer.Peer
+import org.tessellation.sdk.domain.cluster.storage.ClusterStorage
+import org.tessellation.sdk.domain.snapshot.PeerDiscoveryDelay
+import org.tessellation.sdk.infrastructure.snapshot.MajorityPeerSelect.NoPeersToSelect
+
+import eu.timepit.refined.auto._
+import eu.timepit.refined.types.numeric.PosInt
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+import retry.RetryPolicies.{constantDelay, limitRetriesByCumulativeDelay}
+import retry.retryingOnFailures
+
+object SelectablePeerDiscoveryDelay {
+
+  def make[F[_]: Async](
+    clusterStorage: ClusterStorage[F],
+    appEnvironment: AppEnvironment,
+    checkPeersAttemptDelay: FiniteDuration,
+    checkPeersMaxDelay: FiniteDuration,
+    additionalDiscoveryDelay: FiniteDuration,
+    minPeersCount: PosInt
+  ): PeerDiscoveryDelay[F] = new PeerDiscoveryDelay[F] {
+
+    private val logger = Slf4jLogger.getLogger[F]
+
+    private val fallbackPeersCount: PosInt = 1
+
+    private def waitUntil(minPeers: PosInt): F[Set[Peer]] = retryingOnFailures[Set[Peer]](
+      limitRetriesByCumulativeDelay(
+        threshold = checkPeersMaxDelay,
+        policy = constantDelay(checkPeersAttemptDelay)
+      ),
+      a => (a.size >= minPeers).pure[F],
+      (a, details) =>
+        logger.info(
+          s"Discovered ${a.size}/$minPeers selectable peers, waiting $checkPeersAttemptDelay: $details"
+        )
+    )(clusterStorage.getResponsivePeers.map(_.filter(_.state === Ready)))
+
+    def waitForPeers: F[Unit] = {
+      val peers = if (appEnvironment === Dev) {
+        waitUntil(fallbackPeersCount)
+      } else {
+        waitUntil(minPeersCount) >>
+          Async[F].sleep(additionalDiscoveryDelay) >>
+          waitUntil(fallbackPeersCount)
```

Comment:
So, the purpose of

```
waitUntil(minPeersCount) >>
          Async[F].sleep(additionalDiscoveryDelay)
```

is simply to wait for more than 1 peer. Since we don't care about the exact peer count, that has been simplified to just a single delay. Afterwards, we check for 1 peer since `PeerSelect` needs at least 1 peer.

TLDR:

The function has been simplified to

```
    def waitForPeers: F[Unit] =
      Async[F].sleep(additionalDiscoveryDelay).unlessA(appEnvironment === Dev) >>
        waitUntil(fallbackPeersCount).flatMap {
          case s if s.nonEmpty => ().pure
          case _               => NoPeersToSelect.raiseError[F, Unit]
        }
```

Reply:
Per our discussion offline, this has been re-factored ro

```
    def waitForPeers: F[Unit] = {
      val peers =
        if (appEnvironment === Dev)
          waitUntil(fallbackPeersCount)
        else
          waitUntil(minPeers).flatMap {
            case s if s.size >= minPeers => Temporal[F].sleep(additionalDiscoveryDelay).as(s)
            case _ =>
              Temporal[F].sleep(additionalDiscoveryDelay) >> waitUntil(fallbackPeersCount)
          }

      peers.flatMap {
        case s if s.nonEmpty => ().pure
        case _               => NoPeersToSelect.raiseError[F, Unit]
      }
    }
```